### ISSUE TITLE
Error importing filtered particles to cryoSPARC from chunked particle stacks

### ISSUE THREAD

**Issue by priiteek** [zhonge/cryodrgn#88]
I have a larger particle set, so I chunk it to multiple mrcs stacks with cryodrgn downsample. After training the dragon, filtering in jupyter notebook, writing the particles to a star-file with write_starfile.py, and trying to import it in cryoSPARC, it returns an error:
```
Traceback (most recent call last):
  File "cryosparc_master/cryosparc_compute/run.py", line 84, in cryosparc_compute.run.main
  File "/data/software/cryosparc/cryosparc_master/cryosparc_compute/jobs/imports/run.py", line 443, in run_import_particles
    fig = plotutil.plot_images_simple(dispparts, rows=4, cols=8, radwn=6, figscale=1.5)
  File "/data/software/cryosparc/cryosparc_master/cryosparc_compute/plotutil.py", line 495, in plot_images_simple
    rD = img.get_original_real_data()
  File "/data/software/cryosparc/cryosparc_master/cryosparc_compute/particles.py", line 15, in get_original_real_data
    return self.blob.view().copy()
  File "/data/software/cryosparc/cryosparc_master/cryosparc_compute/blobio/mrc.py", line 115, in view
    return self.get()
  File "/data/software/cryosparc/cryosparc_master/cryosparc_compute/blobio/mrc.py", line 110, in get
    _, data, total_time = prefetch.synchronous_native_read(self.fname, idx_start = self.page, idx_limit = self.page+1)
  File "cryosparc_master/cryosparc_compute/blobio/prefetch.py", line 64, in cryosparc_compute.blobio.prefetch.synchronous_native_read
RuntimeError: mrc_readmic (1) 447: Invalid argument
[arguments] <path to cs job>/imported/particles.256.2.mrcs, 0, 0, 104136, 104137, 2, 40
```

I tested this with the tutorial dataset. I ran the training with boxsize 256 particles, which are also chunked into multiple stacks in the tutorial. I picked a random cluster and exported the indices to ind_keep.5899_particles.pkl. To write the starfile:

`python /path/to/cryodrgn/utils/write_starfile.py particles.256.txt ctf.pkl --poses poses.pkl --ind 01_vae256/ind_keep.5899_particles.pkl --full-path -o 01_k13.256.star`

When I tried to import the generated star-file in cryoSPARC, I got the error above. This happens only if the particles are chunked, i.e. the input is a txt file that refers to multiple mrcs stacks. If only a single mrcs is used, everything works as expected. 

Perhaps there is some kind of a mess-up with the indices when writing out the star-file that refers to multiple stacks? It looks like the indices referring to images in subsequent mrcs stacks (1.mrcs, 2.mrcs ...) get really big. Shouldn't the number of particles in the previous chunks be subtracted from the index? E.g. if each chunk contains 100k particles, then the indices of 0.mrcs are what they are, but for 1.mrcs should be i-100k, for 2.mrcs i-200k, etc?

**Comment by zhonge** [comment]
Thank you for finding and reporting this! This is indeed a bug with the indices when converting back to star file with a chunked dataset. 

I will fix this. In the meantime, can you send me an email (zhonge@mit.edu) with the .star file just to double check.

**Comment by priiteek** [comment]
Thanks for working on this. I've sent you the star-file.

**Comment by zhonge** [comment]
Fixed in eeab1446eaa8972b5a089a9428bdaa7e2b259fe6. @priiteek reported that the updated script works.
